4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+//rework enums
+//restore proper list
+
+namespace AnimalShelter.Code
+{
+    public class AnimalShelter
+    {
+        //private List<Animal> _animals = new List<Animal>();
+        private List<Animal> _animals = new List<Animal>();
+        public List<Animal> animals
+        {
+            get { return _animals; }
+            set { _animals = value; }
+        }
+        public AnimalShelter()
+        {
+               
+        }
+        public Result AddAnimal(Animal animal)
+        {
+            if (animal.AnimalType == AnimalType.Deer || animal.AnimalType == AnimalType.Bear)
+            {
+                return new Result( "Animal was not a supported animal");
+            }
+            else
+            {
+                _animals.Add(animal);//using mine
+                return new Result(animal);
+            }
+        }
+
+        public Result RemoveAnimal(Animal animal)
+        {
+            if (animal.AnimalType == AnimalType.Deer || animal.AnimalType == AnimalType.Bear)
+            {
+                return new Result("Animal was not a supported animal");
+            }
```

Comment:
Now that you have used this logic in two areas, this seems like reusable code that could have been in a function somewhere. I could also see this being exposed as a public function. Something then could ask the shelter "Do you support this?"